util/unifycr/src/unifycr-rm.c

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2017, Lawrence Livermore National Security, LLC.
+ * Produced at the Lawrence Livermore National Laboratory.
+ *
+ * Copyright 2017, UT-Battelle, LLC.
+ *
+ * LLNL-CODE-741539
+ * All rights reserved.
+ *
+ * This is the license for UnifyCR.
+ * For details, see https://github.com/LLNL/UnifyCR.
+ * Please read https://github.com/LLNL/UnifyCR/LICENSE for full license text.
+ */
+
+/*
+ * Copyright (c) 2017, Lawrence Livermore National Security, LLC.
+ * Produced at the Lawrence Livermore National Laboratory.
+ * Copyright (c) 2017, Florida State University. Contributions from
+ * the Computer Architecture and Systems Research Laboratory (CASTL)
+ * at the Department of Computer Science.
+ *
+ * Written by: Hyogi Sim
+ * LLNL-CODE-728877. All rights reserved.
+ *
+ * This file is part of burstfs.
+ * For details, see https://github.com/llnl/burstfs
+ * Please read https://github.com/llnl/burstfs/LICENSE for full license text.
+ */
+
+/*
+ *
+ * Copyright (c) 2014, Los Alamos National Laboratory
+ *	All rights reserved.
+ *
+ */
+
+#ifndef _CONFIG_H
+#define _CONFIG_H
+#include <config.h>
+#endif
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <ctype.h>
+#include <errno.h>
+
+#include "unifycr.h"
+
+/**
+ * @brief get the node list from the $PBS_NODEFILE
+ *
+ * @param resource
+ *
+ * @return
+ */
+static int read_resource_pbs(unifycr_resource_t *resource)
+{
+    int ret = 0;
+    int i = 0;
+    int n_nodes = 0;
+    FILE *fp = NULL;
+    char **nodes = NULL;
+    char buf[1024] = { 0, };
+    char *nodefile = getenv("PBS_NODEFILE");
+
+    if (!nodefile)
+        return -EINVAL;
+
+    fp = fopen(nodefile, "r");
+    if (!fp)
+        return -errno;
+
+    while (fgets(buf, 1023, fp))
+        n_nodes++;
+
+    if (ferror(fp)) {
+        ret = -errno;
+        goto out;
+    }
+
+    nodes = calloc(sizeof(char *), n_nodes);
+    if (!nodes) {
+        ret = errno;
+        goto out;
+    }
+
+    rewind(fp);
+
+    while (fgets(buf, 1023, fp) != NULL) {
```

Comment:
This should be sizeof(buf) for the size parameter.  fgets() always returns a null terminated string on success, so the next line is not needed.